components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/UserProfileExportServiceComponent.java

```diff
@@ -0,0 +1,169 @@
+import org.wso2.carbon.consent.mgt.core.ConsentManager;
+import org.wso2.carbon.identity.user.export.core.internal.service.impl.BasicUserInformationProvider;
+import org.wso2.carbon.identity.user.export.core.internal.service.impl.ConsentInformationProvider;
+import org.wso2.carbon.identity.user.export.core.internal.service.impl.SecurityInformationProvider;
+import org.wso2.carbon.identity.user.export.core.internal.service.impl.UserInformationServiceImpl;
+import org.wso2.carbon.identity.user.export.core.service.UserInformationProvider;
+import org.wso2.carbon.identity.user.export.core.service.UserInformationService;
+import org.wso2.carbon.identity.user.export.core.service.impl.LinkedAccountsProvider;
+import org.wso2.carbon.identity.user.export.core.service.impl.UserProfileInformationProvider;
+import org.wso2.carbon.identity.user.profile.mgt.association.federation.FederatedAssociationManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * User profile export service component.
+ */
+@Component(
+        name = "user.profile.export.service",
+        immediate = true
+)
+public class UserProfileExportServiceComponent {
+
+    private static final Log LOG = LogFactory.getLog(UserProfileExportServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext ctxt) {
+
+        try {
+            UserInformationServiceImpl userInformationService = new UserInformationServiceImpl();
+            ctxt.getBundleContext().registerService(UserInformationService.class.getName(), userInformationService,
+                    null);
+
+            UserProfileInformationProvider userProfileInfoProvider = new UserProfileInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), userProfileInfoProvider,
+                    null);
+
+            LinkedAccountsProvider federatedAccountsProvider = new LinkedAccountsProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), federatedAccountsProvider,
+                    null);
+
+            BasicUserInformationProvider basicUserInformationProvider = new BasicUserInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), basicUserInformationProvider,
+                    null);
+
+            ConsentInformationProvider consentInformationProvider = new ConsentInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), consentInformationProvider,
+                    null);
+
+            SecurityInformationProvider securityInformationProvider = new SecurityInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), securityInformationProvider,
+                    null);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
```

Comment:
What will be the e.getMessage(). In other places, we have error msg as `LOG.error("Error while activating UserProfileExportServiceComponent", e);